mitmproxy/addons/next_layer.py

```diff
@@ -184,6 +184,8 @@ def s(*layers):
             len(data_client) < 3
             # HTTP would require whitespace before the first newline
             # if we have neither whitespace nor a newline, it's also unlikely to be HTTP.
+            or (data_client.find(b" ") == -1)
+            or (data_client.find(b"\n") == -1)
```

Comment:
We can have incomplete request lines in the first packet, so I would rather go for something like this:

```suggestion
            or ("\n" in data_client and " " not in data_client)
```

What do you think?

Reply:
That makes sense, thanks!